consensus/src/units/mod.rs

```diff
@@ -200,34 +181,77 @@ pub(crate) type UncheckedSignedUnit<H, D, S> = UncheckedSigned<FullUnit<H, D>, S
 
 pub(crate) type SignedUnit<H, D, K> = Signed<FullUnit<H, D>, K>;
 
-#[derive(Clone, Eq, PartialEq, Debug, Decode, Encode)]
-pub struct Unit<H: Hasher> {
-    pre_unit: PreUnit<H>,
-    hash: H::Hash,
-}
+/// Abstract representation of a unit from the Dag point of view.
+pub trait Unit: 'static + Send + Clone {
```

Comment:
This and `WrappedUnit` seem cool, but what's the reason why you introduce them? Just asking, not questioning.